torch/_dynamo/variables/builtin.py

```diff
@@ -1234,6 +1234,17 @@ def _grad_changed(old, new):
             if is_utils_checkpoint(member):
                 options["source"] = source
                 return build_checkpoint_variable(**options)
+            elif istype(
```

Comment:
This is a good point that I want some suggestion. Currently I use `istype` to keep sync with checks elsewhere, e.g. constant check in this pr in guards.py

Reply:
@Skylion007 Since the new `common_constant_types` set, I've replace this with an `istype` check for common constants. Although a subtyping check could be more general, but it's quite a strong assumption that all user-defined types work well if with a subclass check here.